src/components/class-management/class-management.tsx

```diff
@@ -0,0 +1,89 @@
+import { useContext, useEffect, useState } from "react";
+import { Button, Form, Input, message, Table } from "antd";
+import axios from "axios";
+import { AuthContext } from "../../state/reducer";
+import "./class-management.scss";
+
+interface IClass {
+  internalName: string;
+  displayedName: string;
+}
+
+function ClassManagement() {
+  const [classes, setClasses] = useState<IClass[]>([]);
+  const [loading, setLoading] = useState(false);
+  const { token } = useContext(AuthContext);
+
+  useEffect(() => {
+    setLoading(true);
+    axios
+      .get("/classes")
+      .then((response) => {
+        setClasses(response.data.classes);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const handleUpdateDisplayedName = (
+    internalName: string,
+    displayedName: string,
+    token: string
+  ) => {
+    setLoading(true);
+    axios
+      .put(`/classes/${encodeURIComponent(internalName)}`, null, {
+        params: {
+          displayed_name: displayedName,
+        },
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+      .then(() => {
+        message.success("Displayed name updated successfully");
+      })
+      .catch((error) => {
+        console.error(error);
+        message.error("Displayed name update failed");
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const columns = [
+    {
+      title: "Internal Name",
+      dataIndex: "internalName",
+      key: "internalName",
+    },
+    {
+      title: "Displayed Name",
+      dataIndex: "displayedName",
+      key: "displayedName",
+      render: (text: string, record: IClass) => (
+        <Input
+          defaultValue={text}
+          onBlur={(e) =>
+            handleUpdateDisplayedName(record.internalName, e.target.value, token)
```

Comment:
You have a type error here, if the token is `null`, then you should redirect the user out of here.

Reply:
man do this please, I dont know why it dosent work with me